advanced/Scripts/piholeDebug.sh

```diff
@@ -9,534 +9,1071 @@
+    done
+    # Print a blank line just for formatting
+    log_write ""
   else
-    log_echo "No IPv${protocol} found on ${PIHOLE_INTERFACE}"
+    # If there are no IPs detected, explain that the protocol is not configured
+    log_write "${CROSS} ${COL_LIGHT_RED}No IPv${protocol} address(es) found on the ${PIHOLE_INTERFACE}${COL_NC} interace.\n"
     return 1
   fi
+  # If the protocol is v6
+  if [[ "${protocol}" == "6" ]]; then
+    # let the user know that as long as there is one green address, things should be ok
+    log_write "   ^ Please note that you may have more than one IP address listed."
+    log_write "   As long as one of them is green, and it matches what is in ${PIHOLE_SETUP_VARS_FILE}, there is no need for concern.\n"
+    log_write "   The link to the FAQ is for an issue that sometimes occurs when the IPv6 address changes, which is why we check for it.\n"
+  fi
 }
 
-ip_ping_check() {
-  local protocol=${1}
-  local cmd
-
+ping_ipv4_or_ipv6() {
+  # Give the first argument a readable name (a 4 or a six should be the argument)
+  local protocol="${1}"
+  # If the protocol is 6,
   if [[ ${protocol} == "6" ]]; then
+    # use ping6
     cmd="ping6"
-    g_addr="2001:4860:4860::8888"
+    # and Google's public IPv6 address
+    public_address="2001:4860:4860::8888"
   else
+    # Otherwise, just use ping
     cmd="ping"
-    g_addr="8.8.8.8"
+    # and Google's public IPv4 address
+    public_address="8.8.8.8"
   fi
+}
 
-  local ip_def_gateway=$(ip -${protocol} route | grep default | cut -d ' ' -f 3)
-  if [[ -n ${ip_def_gateway} ]]; then
-    echo -n ":::        Pinging default IPv${protocol} gateway: "
-    if ! ping_gateway="$(${cmd} -q -W 3 -c 3 -n ${ip_def_gateway} -I ${PIHOLE_INTERFACE} | tail -n 3)"; then
-     log_echo "Gateway did not respond."
-     return 1
-    else
-      log_echo "Gateway responded."
-      log_write "${ping_gateway}"
-    fi
-    echo -n ":::        Pinging Internet via IPv${protocol}: "
-    if ! ping_inet="$(${cmd} -q -W 3 -c 3 -n ${g_addr} -I ${PIHOLE_INTERFACE} | tail -n 3)"; then
-      log_echo "Query did not respond."
+ping_gateway() {
+  local protocol="${1}"
+  ping_ipv4_or_ipv6 "${protocol}"
+  # Check if we are using IPv4 or IPv6
+  # Find the default gateway using IPv4 or IPv6
+  local gateway
+  gateway="$(ip -${protocol} route | grep default | cut -d ' ' -f 3)"
+
+  # If the gateway variable has a value (meaning a gateway was found),
+  if [[ -n "${gateway}" ]]; then
+    log_write "${INFO} Default IPv${protocol} gateway: ${gateway}"
+    # Let the user know we will ping the gateway for a response
+    log_write "   * Pinging ${gateway}..."
+    # Try to quietly ping the gateway 3 times, with a timeout of 3 seconds, using numeric output only,
+    # on the pihole interface, and tail the last three lines of the output
+    # If pinging the gateway is not successful,
+    if ! ${cmd} -c 3 -W 2 -n ${gateway} -I ${PIHOLE_INTERFACE} >/dev/null; then
```

Comment:
Will you send me the `bash -x` output from that section of the log after running the script?  Feel free to send it to `tricorder` so we don't clutter up this comment thread.